sessionctx/variable/variable.go

```diff
@@ -132,6 +132,9 @@ type SysVar struct {
 	GetSession func(*SessionVars) (string, error)
 	// GetGlobal is a getter function for global scope.
 	GetGlobal func(*SessionVars) (string, error)
+	// GetStateValue gets the value for session states, which is used for migrating sessions.
+	// We need a function to override GetSession sometimes, because GetSession may not return the real value.
+	GetStateValue func(*SessionVars) (string, bool, error)
```

Comment:
It's weird, because it may not be an error when `keep==false`.
E.g. When `timestamp` is never set, we don't keep it, but it's not an error, it's normal.

Reply:
Well, I think it is good to use `errors.Is` or so to judge errors. But if you like an additional variable...

Reply:
> GetSession may not return the real value

Could you give an example that 'real value' is required for migration?

Reply:
Examples are given in the test cases.
E.g. `select @@rand_seed1` always returns 0, but the real value is not 0.

Reply:
Got it, thanks.